packages/model-viewer/src/features/environment.ts

```diff
@@ -113,7 +144,41 @@ export const EnvironmentMixin = <T extends Constructor<ModelViewerElementBase>>(
       }
     }
 
-    async[$updateEnvironment]() {
+    [$applyLightProbe]() {
+      const coefficients = this.sphericalHarmonics?.split(' ').map(parseFloat);
```

Comment:
Parsing `coefficients` should be more robust including the `for` loop below. This functionality likely should live in `packages/model-viewer/src/styles/parsers.ts`. I can move this functionality there if we'd like to move ahead with this functionality.